book/src/tooling/coap.md

```diff
@@ -0,0 +1,242 @@
+
+[CoAP] is an **application layer protocol similar to HTTP** in its use
+(you can POST data onto a resource on a server that is identified by a URL)
+but geared towards IoT devices in its format and its security mechanisms.
+
+CoAP provides a versatile set of transports
+(with IETF Proposed Standards for running [over UDP] including multicast, [over TCP and WebSockets],
+other standards for running [over SMS and NB-IoT], and more in development).
+It relies on proxies to span across transports and accommodate the characteristics of particular networks,
+and offers features exceeding the classical REST set such as [observation].
+
+**RIOT-rs supports** the use of CoAP
+for implementing clients, servers or both in a single device.
+As there is more complexity in a CoAP client than a CoAP server,
+the most minimal use is to just have a CoAP server on the embedded system.
+Having both components in use is also a common choice.
+
+As part of our mission for strong security,
+we use encrypted CoAP traffic by default.
+Of the multiple means to secure CoAP,
+we have picked the OSCORE/EDHOC security mechanisms as our default,
+with the option to use ACE for authorization management.
+Alternatives (DTLS, TLS, IPsec or link-layer encryption)
+are currently not implemented.
+
+[CoAP]: https://coap.space/
+[over UDP]: https://datatracker.ietf.org/doc/html/rfc7252
+[over TCP and WebSockets]: https://datatracker.ietf.org/doc/html/rfc8323
+[over SMS and NB-IoT]: https://www.omaspecworks.org/wp-content/uploads/2018/10/Whitepaper-11.1.18.pdf
+[observation]: https://datatracker.ietf.org/doc/html/rfc7641
+
+## Usage: Server side
+
+An example of a CoAP server is [provided as "examples/coap"], see [its `run()` function] for the practical steps.
+
+A CoAP server is created by assembling several **resource handlers on dedicated paths**:
+There might be a path `/s/0` representing a particular sensor,
+and a path `/fwup` for interacting with firmware updates.
+The CoAP implementation can put additional resources at well-known locations,
+eg. `/.well-known/core` for discovery or `/.well-known/edhoc` for establishing secure connections.
+
+The handler does not need to concern itself with authorization:
+The CoAP stack is configured with an access policy,
+which is evaluated before the handler even receives the request.
+The handler still needs to concern itself with security to some extent
+(eg. file format parsers should still treat incoming data as possibly malformed),
+but is not the main enforcement point.
+
+Policy is configured depending on the enabled security mechanisms.
+Examples of described policy entries are:
+
+* This is an fixed public key, and requests authenticated with that key are allowed to GET and PUT to `/limit`.
+* The device has a shared secret from its authorization server, and any requests coming along a token from that server may do whatever the token says they may do (eg. "GET `/limit` and PUT `/led/0`").
```

Comment:
This sentence is a bit confusing to me, especially the second half.

Reply:
Good point, rephrased in 01998806decfd9b7e218e58751619d446e14dd91:

> The device has a shared secret from its authorization server, with which the authorization server secures the tokens it issues to clients. Clients may perform any action as long as they securely present a token that allows it. For example, a token may allow GET on `/limit` and PUT on `/led/0`".